sklearn/manifold/t_sne.py

```diff
@@ -707,22 +718,26 @@ def _fit(self, X, skip_num_points=0):
             raise ValueError("n_iter should be at least 250")
 
         n_samples = X.shape[0]
-
         neighbors_nn = None
+
         if self.method == "exact":
             # Retrieve the distance matrix, either using the precomputed one or
             # computing it.
             if self.metric == "precomputed":
                 distances = X
             else:
+                metric_params_ = ({} if (self.metric_params is None)
+                                  else self.metric_params)
                 if self.verbose:
                     print("[t-SNE] Computing pairwise distances...")
 
                 if self.metric == "euclidean":
```

Comment:
As you can see here, with `metric='euclidean'` the distances are squared.
To keep current behavior, you need to have the distances squared also when `metric='minkowski'` and `p=2`.
In the future, we might keep the distances squared for all metrics (see #12419), but for now we need to have the same behavior than before.
This also needs an update in the case `method='barnes-hut'`, line ~785.

Note also the discussion about using minkowski by default in https://github.com/scikit-learn/scikit-learn/issues/12437#issuecomment-432177619.